libraries/shared/watcher/storage_watcher_test.go

```diff
@@ -196,7 +196,27 @@ func SharedExecuteBehavior(input *ExecuteInput) {
 			err := storageWatcher.Execute()
 
 			Expect(err).To(HaveOccurred())
-			Expect(err).To(MatchError(sql.ErrNoRows))
+			Expect(err).To(MatchError(fakes.FakeError))
+			Expect(mockDiffsRepository.MarkTransformedPassedID).NotTo(Equal(diffWithoutHeader.ID))
+		})
+
+		It("does not return error if transformer fails execution bc of ErrNoRows getting header", func() {
+			diffWithoutHeader := types.PersistedDiff{
+				RawDiff: types.RawDiff{
+					Address:     contractAddress,
+					BlockHash:   test_data.FakeHash(),
+					BlockHeight: rand.Int(),
+				},
+				ID: rand.Int63(),
+			}
+			setDiffsToReturn(storageWatcher.DiffStatus, mockDiffsRepository, []types.PersistedDiff{diffWithoutHeader})
+			setGetDiffsErrors(storageWatcher.DiffStatus, mockDiffsRepository, []error{nil, fakes.FakeError})
+			mockHeaderRepository.GetHeaderByBlockNumberError = sql.ErrNoRows
+
+			err := storageWatcher.Execute()
+
+			Expect(err).To(HaveOccurred())
```

Comment:
Confused by this test - you're saying you expect an error to have occurred, but the description says "does not return error". Maybe the description needs to be clarified?

Reply:
good call! updated the description ~

these tests are a little funky bc the watcher is designed to ignore certain classes of errors (like the ones under test here) - so we need to trigger an unexpected error to kill the function. I think we had originally done some async assertions to avoid this step, but that added a lot of complexity. Would welcome thoughts on how to clarify intent - maybe even a comment somewhere?

Reply:
good call! updated the description ~

these tests are a little funky bc the watcher is designed to ignore certain classes of errors (like the ones under test here) - so we need to trigger an unexpected error to kill the function. I think we had originally done some async assertions to avoid this step, but that added a lot of complexity. Would welcome thoughts on how to clarify intent - maybe even a comment somewhere?